### DISCUSSION TITLE
Passive Transport between two compartments with similar structure

### DISCUSSION THREAD

**Discussion by moPellowe** [Open-Systems-Pharmacology/Forum#1114] [Questions & Problems] [answered]
I'm wanting to have a Passive Transport defined between ExtracellularFluid compartments within different layers (see diagram below). How can I define the source and target tags to allow for this? I know that I could simply say "In container layer 1" for the source and "In container layer 2" for target, but I'm wanting to make many, many layers, and it would be ideal to have a single Passive Transport to apply to all of them. Does anyone have any ideas on how I could do this efficiently (instead of the brute force way, i.e. a separate Passive Transport for each neighborhood)?

![image](https://user-images.githubusercontent.com/57540473/152622592-726627b7-a766-4035-a2c2-729e522761c9.png)

**Comment by Yuri05** [comment]
So you want to have the transports 
`Layer1/ExtracellularFluid`<=>`Layer2/ExtracellularFluid`<=>`Layer3/ExtracellularFluid`<=>... ?

~It's enough to set **SourceCriteria = "ExtracellularFluid"** and **TargetCriteria = "ExtracellularFluid"**~
Passive transports are only created if there is a neighborhood between 2 containers, so as long as you have a neighborhood between 2 extracellular fluid compartments of 2 adjacent layers - it's fine.

It will become problematic, if you have a neighborhood between 2 ExtracellularFluid compartments where you **don't want** to have a transport (e.g. additionally to `Layer1/ExtracellularFluid` <=>` Layer2/ExtracellularFluid` you have a neighborhood `Layer1/ExtracellularFluid` <=> `SomethingElse/ExtracellularFluid `). Is this the case?

EDIT: no, this would not work, because it's not clear what is the source and the target...

**Comment by Yuri05** [comment, marked as answer]
I don't see a possibility to solve it with only 1 transport. 
You will need to create N transports. They all can share the same formula, so you don't need N formulas at least.

P.S. transferring the issue to the forum.

**Reply by moPellowe** [reply]
Is there a way to do this more programmatically, either within MoBi or with the raw files?

**Reply by msevestre** [reply]
You could export the pkml file for passsive transport....create one or two by hand and then see how the file structure looks like and add some more by script.
But there is nothing that Mobi is supporting for this. The likelihood that you will corrupt the pkml is quite high....

**Reply by moPellowe** [reply]
I see. So do at your own risk. ;)  Thank you @Yuri05 and @msevestre for your input.

**Reply by msevestre** [reply]
Yes. That being said, if the pkml is not correct, Mobi won't be able to load it...make sure specifically that id are all unique when they need to be 
Best would be to create 3 by hand referencing the same formula....and then you can duplicate
